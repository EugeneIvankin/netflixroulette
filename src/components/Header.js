import React, { Component } from 'react'
import '../styles/header.css'


export default class Header extends Component {



    searchClick = () => { // прочитать bind, call apply
        this.props.getMoviesByName(this.input.value);
        this.props.history.push(`/search/${this.input.value}`);
    };

    // прочитать про инпут, сделать форм

    //убарть ref, создав контролируемый компонент

    render() {
        return <header>
            <div className="divLogo">
                <p className="logo">netflixroulette</p>
            </div>
            <h3>FIND YOUR MOVIE</h3>
            <input placeholder="Enter your movie" name="title"  ref={node => {this.input = node;}}></input>
            <div className="searchDiv">
                <h5>SEARCH BY</h5>
                <button id="title"><h5 className="textButton">TITLE</h5></button>
                <button className="searchButton" onClick={this.searchClick} ><h3 className="textSearchButton">SEARCH</h3></button>
            </div>
        </header>
    }
}

