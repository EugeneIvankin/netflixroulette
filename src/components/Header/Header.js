import React, { Component } from 'react'
import './header.css'


export default class Header extends Component {

    constructor(props){
        super(props);
        this.state = {value:''};
    }

    handleChange = (event) =>{
        this.setState({value: event.target.value})
    }

    handleSubmit = (event) =>{
        event.preventDefault();
        this.props.history.push(`/search/${this.state.value}`);
    }

    render() {
        return <header>
            <div className="div-logo">
                <p className="logo">netflixroulette</p>
            </div>
            <h3>FIND YOUR MOVIE</h3>
            <form className="search-div" onSubmit={this.handleSubmit}>
                <input type="text" name="movie" value={this.state.value} onChange={this.handleChange} placeholder="Enter your movie"></input>
                <button type="submit">SEARCH</button>
            </form>
        </header>
    }

    //разкинуть тесты
}
