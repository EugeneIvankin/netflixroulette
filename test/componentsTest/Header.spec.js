import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../src/components/Header/Header'


describe('<Header/>', () => {

    it ('should render Header', () => {
        const renderedComponent = shallow(
            <Header/>
        );
       expect(renderedComponent).toMatchSnapshot();
    });

    it ('should call handleChange on input changes', () => {
        const history = {
            push: jest.fn()
        };

        const renderedComponent = shallow(
            <Header history={history}/>);

        renderedComponent.find('input').simulate('change', { target: { value: 'Test' }});
        renderedComponent.find('.searchDiv').simulate('submit', {preventDefault:jest.fn()});
        expect(history.push).toBeCalledWith('/search/Test');
    })

});
