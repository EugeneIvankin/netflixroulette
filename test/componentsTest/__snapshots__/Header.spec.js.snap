// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render Header 1`] = `
<header>
  <div
    className="div-logo"
  >
    <p
      className="logo"
    >
      netflixroulette
    </p>
  </div>
  <h3>
    FIND YOUR MOVIE
  </h3>
  <form
    className="search-div"
    onSubmit={[Function]}
  >
    <input
      name="movie"
      onChange={[Function]}
      placeholder="Enter your movie"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      SEARCH
    </button>
  </form>
</header>
`;
